src/daemon/daemon.cpp

```diff
@@ -2305,7 +2325,7 @@ error_string mp::Daemon::async_wait_for_ssh_and_start_mounts_for(const std::stri
                 server->Write(reply);
             }
 
-            mp::utils::wait_for_cloud_init(vm.get(), cloud_init_timeout, *config->ssh_key_provider);
+            mp::utils::wait_for_cloud_init(vm.get(), timeout, *config->ssh_key_provider);
```

Comment:
So there is one other thing we could do that may be best, and that is respect the `--timeout` in the client too — rather than only in the daemon. We'd still pass the `--timeout` value to the daemon so that the above operations don't time out before the client decides to time out. But the client will still bail if the total time for the operation exceeds `--timeout`. That may well be the most intuitive solution.

Reply:
Hey, so the timeout would be primarily a client thing, the daemon using it only as a high bound for the async operations? Sounds quite reasonable to me. There may be ongoing operations that carry on after the client quits, but I guess docs can have wording that is compatible with that (e.g. "maximum time to wait for ...").

Reply:
Hey, exactly - and leave us free to extend the allowed `--timeout` values in the future, while keeping the most intuitive way just "--timeout <seconds>". I'll have questions on how you guys think it'd be best to achieve.

Reply:
OK so this is done now. Thanks for pushing!

```plain
$ time multipass launch --timeout 10
Timed out waiting for instance launch.                                          

real    0m10,077s
user    0m0,042s
sys     0m0,047s
```